ttnn/cpp/ttnn/operations/data_movement.hpp

```diff
@@ -10,13 +10,55 @@
 #include "ttnn/deprecated/tt_dnn/op_library/composite/composite_ops.hpp"
 #include "ttnn/operations/upsample/upsample_op.hpp"
 #include "ttnn/operations/core/core.hpp"
+#include "ttnn/operations/data_movement/permute/permute.hpp"
 
 #include <ranges>
 
 namespace ttnn {
 namespace operations {
 namespace data_movement {
 
+// repeat interleave supports repeats as 1 to inf, dim between 0 to 2
+inline Tensor repeat_interleave(const Tensor& input_a, uint32_t repeat, int32_t dim, const std::optional<MemoryConfig>& output_mem_config) {
```

Comment:
no no. please place in https://github.com/tenstorrent/tt-metal/tree/main/ttnn/cpp/ttnn/operations/data_movement as a separate folder `repeat_interleave` and follow the proper op structure as described in the ticket / docs